File: sources/01-searching-algorithms/01-sequential-search/c/Main.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

/**
 * Código fonte - Implementação de Pesquisa Sequencial
 *
 * @author Artur Bomtempo Colen
 * @algorithm Sequential Search
 * @type Search Algorithm
 * @language C
 * @version 1.0
 */

#define SIZE 10

int sequentialSearch(int array[], int size, int target) {
    for (int i = 0; i < size; i++) {
        if (array[i] == target) {
            return i;
        }
    }

    return -1;
}

void generateRandomArray(int array[], int size) {
    srand(time(NULL));

    for (int i = 0; i < size; i++) {
        array[i] = rand() % 100 + 1;
    }
}

void printArray(int array[], int size) {
    printf("[ ");

    for (int i = 0; i < size; i++) {
        printf("%d ", array[i]);
    }

    printf("]\n");
}

int main() {
    int array[SIZE];
    generateRandomArray(array, SIZE);
    printArray(array, SIZE);

    int searchTargets[] = {30, 50, 75, 15};
    int numTests = sizeof(searchTargets) / sizeof(searchTargets[0]);

    for (int i = 0; i < numTests; i++) {
        int target = searchTargets[i];
        printf("\nProcurando pelo valor %d...\n", target);
        
        int result = sequentialSearch(array, SIZE, target);

        if (result != -1) {
            printf("Valor %d encontrado no índice %d.\n", target, result);
        } else {
            printf("Valor %d não encontrado no vetor.\n", target);
        }
    }

    return 0;
}

File: sources/01-searching-algorithms/01-sequential-search/java/Main.java
import java.util.Random;

/**
 * Código fonte - Implementação de Pesquisa Sequencial
 *
 * @author Artur Bomtempo Colen
 * @algorithm Sequential Search
 * @type Search Algorithm
 * @language Java
 * @version 1.0
 */

public class Main {
    public static int sequentialSearch(int[] array, int target) {
        for (int i = 0; i < array.length; i++) {
            if (array[i] == target) {
                return i;
            }
        }

        return -1;
    }

    public static void generateRandomArray(int[] array) {
        Random rand = new Random();

        for (int i = 0; i < array.length; i++) {
            array[i] = rand.nextInt(100) + 1;
        }
    }

    public static void printArray(int[] array) {
        System.out.print("[ ");

        for (int num : array) {
            System.out.print(num + " ");
        }

        System.out.println("]");
    }

    public static void main(String[] args) {
        int[] array = new int[10];
        generateRandomArray(array);
        printArray(array);

        int[] searchTargets = { 30, 50, 75, 15 };

        for (int target : searchTargets) {
            System.out.println("\nProcurando pelo valor " + target + "...");
            int result = sequentialSearch(array, target);

            if (result != -1) {
                System.out.println("Valor " + target + " encontrado no índice " + result + ".");
            } else {
                System.out.println("Valor " + target + " não encontrado no vetor.");
            }
        }
    }
}